packages/@aws-cdk/aws-kms/test/key.test.ts

```diff
@@ -582,6 +582,321 @@ describe('imported keys', () => {
   });
 });
 
+describe('fromCfnKey()', () => {
+  let stack: cdk.Stack;
+  let cfnKey: kms.CfnKey;
+  let key: kms.IKey;
+
+  beforeEach(() => {
+    stack = new cdk.Stack();
+    cfnKey = new kms.CfnKey(stack, 'CfnKey', {
+      keyPolicy: {
+        Statement: [
+          {
+            Action: 'kms:*',
+            Effect: 'Allow',
+            Principal: {
+              AWS: cdk.Fn.join('', [
+                'arn:',
+                cdk.Aws.PARTITION,
+                ':iam::',
+                cdk.Aws.ACCOUNT_ID,
+                ':root',
+              ]),
+            },
+            Resource: '*',
+          },
+        ],
+        Version: '2012-10-17',
+      },
+    });
+    key = kms.Key.fromCfnKey(cfnKey);
+  });
+
+  test("correctly resolves the 'keyId' property", () => {
+    expect(stack.resolve(key.keyId)).toStrictEqual({
+      Ref: 'CfnKey',
+    });
+  });
+
+  test("correctly resolves the 'keyArn' property", () => {
+    expect(stack.resolve(key.keyArn)).toStrictEqual({
+      'Fn::GetAtt': ['CfnKey', 'Arn'],
+    });
+  });
+
+  test('preserves the KMS Key resource', () => {
+    expectCdk(stack).to(haveResource('AWS::KMS::Key', {
```

Comment:
I don't think this assertion is sufficient to verify the behaviour described. maybe a combination of checking the logical ID was unchanged (`beASupersetOfTemplate`?) and counting the number of Keys (`countResources`)

Reply:
Can you explain why is this assertion not sufficient? Are you worried there might be more than one Key with this exact same `KeyPolicy`?

Reply:
And is an assertion like `expect(stack).to(countResources('AWS::KMS::Key', 1))` enough to alleviate that concern, or do you think something more is required?

Reply:
Works for me!

Reply:
Done!